src/pcapi/serialization/spec_tree.py

```diff
@@ -38,3 +41,35 @@ def _generate_spec(self) -> dict:
                     schema["paths"][path][method.lower()]["security"] = func.requires_authentication
 
         return schema
+
+
+class ExtendedResponse(SpectreeResponse):
+    def __init__(
+        self,
+        *codes,
+        code_descriptions=None,
+        **code_models,
+    ):
+        self.code_descriptions = code_descriptions or {}
+        super().__init__(*codes, **code_models)
+
+    def get_code_description(self, code):
+        return self.code_descriptions[code] if code in self.code_descriptions else DEFAULT_CODE_DESC[code]
```

Comment:
(style / détail)

en plus concis : 

```suggestion
        return self.code_descriptions.get(code, DEFAULT_CODE_DESC[code])
```

`DEFAULT_CODE_DESC` a l'air de contenir tous les codes HTTP, donc ça me semble équivalent.